### PR TITLE
Use symfony contracts

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is a BC break, 'cause we changed the parent class of `BlockEvent`.

## Changelog

```markdown
### Changed
- Use symfony event dispatcher contracts 
### Removed
- Removed `symfony/event-dispatcher` dependency
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
         "symfony/console": "^2.8 || ^3.2 || ^4.0",
         "symfony/dependency-injection": "^2.8 || ^3.2 || ^4.0",
         "symfony/event-dispatcher": "^2.8 || ^3.2 || ^4.0",
```
Comment (phansys) [inline composer.json]: With these changes, there are no usages left from `Symfony\Component\EventDispatcher\` namespace.
Should not we remove the dependency too?

Comment (core23) [reply composer.json]: Not sure about this. We would remove an implementation for this thing: /

Comment (greg0ire) [reply composer.json]: I think people using it should require it. IMO we should remove it.

File: composer.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
         "symfony/config": "^2.8 || ^3.2 || ^4.0",
         "symfony/console": "^2.8 || ^3.2 || ^4.0",
         "symfony/dependency-injection": "^2.8 || ^3.2 || ^4.0",
-        "symfony/event-dispatcher": "^2.8 || ^3.2 || ^4.0",
```
Comment (OskarStark) [inline composer.json]: Wouldn’t this drop support for SF 2 and 3 ?

Comment (core23) [reply composer.json]: Refs #592 592

**Pull request merged**